iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2386,7 +2397,8 @@ static void SubscribeToMqttProtocol(PMQTTTRANSPORT_HANDLE_DATA transport_data)
                 // Setting the value to 0 as it is simpler than calculating the amount of time to
                 // expire the PUBLISH. This value of zero is equivalent to setting a time way in the 
                 // past, enough for this control logic.
-                msg_detail_entry->msgPublishTime = 0;        // force the message to resend
+                msg_detail_entry->msgPublishTime = 0;       // force the message to resend
+                msg_detail_entry->isReconnectRetry = true;
```

Comment:
ugh, I want to get rid of isReconnectRetry flag......

if we reset mqttMsgEntry->retryCount to zero, and msg_detail_entry->reconnectRetryCount++ here

than we can let publishTelemetryMsg() and ProcessPendingTelemetryMessages() do it's stuff without any code changes to increment reconnectRetryCount. Thoughts? Maybe talk to JohnS.

Reply:
I agree - it would be nice to be able to do without isReconnectRetry. The problem I found is that I need a way to know whether to increment the timeoutRetry or not - if I leave those blocks of code as is and increment the retry counter here, it would increment both counters on reconnect (plus I figured it was better to increment when the message is actually being sent in case there is some issue between the reconnect and send).